core/dbt/events/base_types.py

```diff
@@ -88,6 +90,21 @@ def get_pid(self) -> int:
             self.pid = os.getpid()
         return self.pid
 
+    def get_node_info(self) -> Any:
+        if self.report_node_data:
+            return {
+                "type": 'node_status',
+                "node_path": self.report_node_data.path,
+                "node_name": self.report_node_data.name,
+                "resource_type": self.report_node_data.resource_type,
+                "node_materialized": self.report_node_data.config.materialized,
+                "node_started_at": "TODO",
+                "unique_id": self.report_node_data.unique_id,
+                "node_finished_at": "TODO",
+                "node_status": self.node_status,
+                "run_state": self.node_state
+            }
```

Comment:
I'd rather see this as a class definition than a dictionary so we can dump the `Any` type up top.